lib/saluki-components/src/sources/checks/check_metric.rs

```diff
@@ -0,0 +1,113 @@
+use std::time::Duration;
+
+use saluki_context::{
+    tags::{Tag, TagSet},
+    Context,
+};
+use saluki_core::data_model::event::metric::{Metric, MetricMetadata, MetricValues};
+use saluki_core::data_model::event::Event;
+use tracing::warn;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum MetricType {
+    Gauge = 0,
+    Rate,
+    Count,
+    MonotonicCount,
+    Counter,
+    Histogram,
+    Historate,
+}
+
+impl From<i32> for MetricType {
+    fn from(v: i32) -> Self {
+        match v {
+            0 => MetricType::Gauge,
+            1 => MetricType::Rate,
+            2 => MetricType::Count,
+            3 => MetricType::MonotonicCount,
+            4 => MetricType::Counter,
+            5 => MetricType::Histogram,
+            6 => MetricType::Historate,
+            _ => {
+                warn!("Unknown metric type: {}, considering it as a gauge", v);
+                MetricType::Gauge
+            }
+        }
+    }
+}
+
+/// CheckMetric are used to transmit metrics from python check execution results
+/// to forward in the saluki's pipeline.
+#[derive(Debug, PartialEq)]
+pub struct CheckMetric {
+    name: String,
+    metric_type: MetricType,
+    value: f64,
+    tags: Vec<String>,
+}
+
+impl CheckMetric {
+    pub fn new(name: String, metric_type: MetricType, value: f64, tags: Vec<String>) -> Self {
+        Self {
+            name,
+            metric_type,
+            value,
+            tags,
+        }
+    }
+}
+
+impl From<CheckMetric> for Event {
+    fn from(check_metric: CheckMetric) -> Self {
+        let tags: Vec<Tag> = check_metric.tags.into_iter().map(Tag::from).collect();
+
+        // Convert Vec<Tag> to TagSet
+        let tagset: TagSet = tags.into_iter().collect();
+
+        let context = Context::from_parts(check_metric.name, tagset);
+        let metadata = MetricMetadata::default();
+
+        match check_metric.metric_type {
+            MetricType::Gauge => Event::Metric(Metric::from_parts(
+                context,
+                MetricValues::gauge(check_metric.value),
+                metadata,
+            )),
```

Comment:
Probably a little cleaner to do something like:

```rust
let values = match check_metric.metric_type {
    MetricType::Gauge => MetricValues::gauge(check_metric.value),
    ...
}:

Event::Metric(Metric::from_parts(context, values, metadata))
```

Reply:
Done https://github.com/DataDog/saluki/pull/711/commits/c5233ba99425a31001a200bc781b9cd36ea9e22a